### PR TITLE
Some agent-install.sh user experience improvements

### PR DESCRIPTION
**Pull request by t-fine**
Updates in this PR are the checked boxes in this issue:
https://app.zenhub.com/workspaces/open-horizon-5877f2c6a8ad090f31c73853/issues/open-horizon/anax/1502

- removed creation of `~/.hzn/hzn.json` for Mac around line 542 
- added usage description of `-l`
- moved agent-install.crt to `/etc/horizon` ...this had to be `/private/etc/horizon` for Mac because docker on Mac does not have access to `/etc` by default, but it does have access to `/private` which is a symbolic link to `/etc` 
- removed ` Do you want to overwrite the current node configuration?[y/N]: `
- when rerunning agent-install.sh with the same install pkgs, it won't install them, unless a force flag is specified
- /agent-install.sh -h will exit code of 0 (if the usage is shown as a result of invalid args, then the exit code should be 1)
- getting node IP address with `ipconfig getifaddr en0` on Mac
- added the -z flag to the usage msg
- changed all occurrences of set +x to { set +x; } 2>/dev/null so it doesn't get printed

### CODE REVIEW

File: agent-install/agent-install.sh [hunk 1]
```diff
@@ -1317,14 +1271,8 @@ function check_node_state() {
...
 			log_notify "Your node is registered"
-			echo "Do you want to overwrite the current node configuration?[y/N]:"
-			read RESPONSE
-			if [ "$RESPONSE" == 'y' ]; then
-				OVERWRITE_NODE=true
-				log_notify "The configuration will be overwritten..."
-			else
-				log_notify "You might be asked for overwrite confirmations later..."
-			fi
+			VERWRITE_NODE=true
```
Comment (dabooz) [inline agent-install/agent-install.sh]: This look like a typo?

File: agent-install/agent-install.sh [hunk 2]
```diff
@@ -393,6 +392,30 @@ function version_gt() {
 	test "$(printf '%s\n' "$@" | sort -V | head -n 1)" != "$1";
 }
 
+# create /etc/default/horizon file for mac or linux
+function create_config() {
+    if [[ ! -z "${HZN_EXCHANGE_URL}" ]] && [[ ! -z "${HZN_FSS_CSSURL}" ]]; then
```
Comment (bmpotter) [inline agent-install/agent-install.sh]: use `-n` instead of `! -z`

File: agent-install/agent-install.sh (hunk 1 above)
Comment (bmpotter) [inline agent-install/agent-install.sh]: typo, should be `OVERWRITE_NODE`

Comment (t-fine) [reply agent-install/agent-install.sh]: Fixed

File: agent-install/agent-install.sh (hunk 2 above)
Comment (t-fine) [reply agent-install/agent-install.sh]: Changed all instances of this

File: agent-install/agent-install.sh (hunk 1 above)
Comment (t-fine) [reply agent-install/agent-install.sh]: Fixed

File: agent-install/agent-install.sh [hunk 3]
```diff
@@ -393,6 +392,30 @@ function version_gt() {
 	test "$(printf '%s\n' "$@" | sort -V | head -n 1)" != "$1";
 }
 
+# create /etc/default/horizon file for mac or linux
+function create_config() {
```
[the remark this answers is not in the document]
Comment (t-fine) [reply agent-install/agent-install.sh]: Slacked you about this

**Pull request merged**